src/components/Selectors/BattleTypes.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormControl from '@material-ui/core/FormControl';
+import Icon from '@material-ui/core/Icon';
+import { BATTLETYPES, BATTLETYPES_LONG } from '../../constants/ranking';
+
+const periodTypes = ['all', 'year', 'month', 'week', 'day'];
+
+class BattleTypes extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      type: 'All',
+    };
+  }
+
+  updateType(selectedType) {
+    const { typeUpdated, periodType } = this.props;
+    const { type } = this.state;
+    const types = BATTLETYPES[periodTypes[periodType]];
+    let newType = selectedType;
+    const current = types.indexOf(type);
+    if (selectedType === -1) {
+      if (current - 1 >= 0) {
+        newType = types[current - 1];
+      } else {
+        newType = type;
+      }
+    }
+    if (selectedType === 1) {
+      if (current + 1 < types.length) {
+        newType = types[current + 1];
+      } else {
+        newType = type;
+      }
+    }
+    this.setState({ type: newType });
+    typeUpdated(newType);
```

Comment:
If you need to do something (e.g. call a function) after setState the proper way is to pass it in the callback of the setState function ([https://reactjs.org/docs/react-component.html#setstate](https://reactjs.org/docs/react-component.html#setstate)). Just a heads-up, I've seen a few places in the code where that is probably what's intended.

Reply:
but typeUpdated is a prop that sends the value to the parent, it's not really dependent on the state being updated, it could be placed above setState and still work the same. Of course you could argue that you want to be sure the UI has been updated before the parent gets the value, in this case it will start a new db call so would rather start it as fast as possible, not sure if there would be any possible downsides

Reply:
Most of the time these types of things are not a problem on ui but it leads to a state where in code level the application state is not clear which is not a good practice. Anyway, I just thought I'd mention about it now so people are at least aware of it.